mat64/product.go

```diff
@@ -0,0 +1,190 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mat64
+
+import "fmt"
+
+// Product calculates the product of the given factors and places the result in
+// the receiver. The order of multiplication operations is optimized to minimize
```

Comment:
Can we say something more vague in the comment? For example, if we have a smart algorithm that notes the difference in FLOPS (or clock time) between Gemm and Trmm we shouldn't be constrained not to use it.